qutebrowser/completion/models/sql.py

```diff
@@ -0,0 +1,302 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2016 Ryan Roden-Corrent (rcorre) <ryan@rcorre.net>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The base completion model for completion in the command line.
+
+Module attributes:
+    Role: An enum of user defined model roles.
+"""
+
+import re
+
+from PyQt5.QtCore import Qt, QModelIndex, QAbstractItemModel
+from PyQt5.QtSql import QSqlTableModel, QSqlDatabase, QSqlQuery
+
+from qutebrowser.utils import usertypes, log
+
+
+Role = usertypes.enum('Role', ['sort'], start=Qt.UserRole, is_int=True)
+
+
+def init(path):
+    """Initialize the SQL completion module.
+
+    Args:
+        path: Path to the completion database.
+    """
+    database = QSqlDatabase.addDatabase('QSQLITE')
+    database.setDatabaseName(path)
```

Comment:
We could also use `:memory:` for an [in-memory database](https://sqlite.org/inmemorydb.html), at least if building it is fast enough.